src/crypto/crypto_dh.cc

```diff
@@ -539,62 +539,68 @@ WebCryptoKeyExportStatus DHKeyExportTraits::DoExport(
 }
 
 namespace {
-AllocatedBuffer StatelessDiffieHellman(
-    Environment* env,
-    ManagedEVPPKey our_key,
-    ManagedEVPPKey their_key) {
-  size_t out_size;
-
-  EVPKeyCtxPointer ctx(EVP_PKEY_CTX_new(our_key.get(), nullptr));
+bool CreateStatelessDiffieHellmanContext(
+    const ManagedEVPPKey& our_key,
+    const ManagedEVPPKey& their_key,
+    char** buf,
+    size_t* out_size) {
+  const EVPKeyCtxPointer ctx(EVP_PKEY_CTX_new(our_key.get(), nullptr));
   if (!ctx ||
       EVP_PKEY_derive_init(ctx.get()) <= 0 ||
       EVP_PKEY_derive_set_peer(ctx.get(), their_key.get()) <= 0 ||
-      EVP_PKEY_derive(ctx.get(), nullptr, &out_size) <= 0)
-    return AllocatedBuffer();
+      EVP_PKEY_derive(ctx.get(), nullptr, out_size) <= 0)
+    return false;
 
-  AllocatedBuffer result = AllocatedBuffer::AllocateManaged(env, out_size);
-  CHECK_NOT_NULL(result.data());
+  *buf = MallocOpenSSL<char>(*out_size);
```

Comment:
Where is this released when this function returns `false`?

Reply:
@addaleax good catch, fixed!